### PR TITLE
[FLINK-28758] Fix stop-with-savepoint for FlinkKafkaConsumer

### PR DESCRIPTION
**Pull request by pnowojski**
https://issues.apache.org/jira/browse/FLINK-28758?focusedCommentId=17700124&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17700124

### CODE REVIEW

File: flink-connector-kafka/src/test/resources/log4j2-test.properties [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 # Set root logger level to OFF to not flood build logs
 # set manually to INFO for debugging purposes
-rootLogger.level = OFF
+rootLogger.level = INFO
```
Comment (dawidwys) [inline flink-connector-kafka/src/test/resources/log4j2-test.properties:21]: do we really need that or is it just a leftover?

File: flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerITCase.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+
+/** ITCase tests class for {@link FlinkKafkaConsumer}. */
+public class FlinkKafkaConsumerITCase {
+    private static final String TOPIC1 = "topic1";
+    private static final String TOPIC2 = "topic2";
+
+    @Nested
```
Comment (dawidwys) [inline flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerITCase.java:53]: Do we need those tests to be `Nested`? Can't we have them top-level?

**Review by dawidwys — APPROVED** [review]
2 minor comments, lgtm

File: flink-connector-kafka/src/test/resources/log4j2-test.properties (hunk 1 above)
Comment (pnowojski) [reply flink-connector-kafka/src/test/resources/log4j2-test.properties:21]: left over!

File: flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerITCase.java (hunk 2 above)
Comment (pnowojski) [reply flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerITCase.java:53]: No, good point.

**Review by tzulitai — APPROVED** [review]

**Comment by tzulitai** [comment]
thanks for the fix @pnowojski! Will you merge this to `main` and `v3.0` branches?

**Comment by pnowojski** [comment]
Yes, will do. @tzulitai , after merging this to v3.0, what Flink fix version should I set for the ticket?

**Pull request merged**